treescript/mercurial.py

```diff
@@ -0,0 +1,196 @@
+"""Treescript mercurial functions."""
+# import asyncio
+# from asyncio.subprocess import PIPE, STDOUT
+# import functools
+# import hashlib
+# import json
+import logging
+import os
+import sys
+# from shutil import copyfile
+# import traceback
+# from collections import namedtuple
+
+from treescript.utils import execute_subprocess
+from treescript.exceptions import FailedSubprocess
+from treescript.task import get_source_repo, get_tag_info
+
+# https://www.mercurial-scm.org/repo/hg/file/tip/tests/run-tests.py#l1040
+# For environment vars.
+
+HGRCPATH = os.path.join(os.path.dirname(__file__), 'data', 'hgrc')
+ROBUSTCHECKOUT_PATH = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), 'py2', 'robustcheckout.py')
+)
+TAG_MSG = "No bug - Tagging {revision} with {tags} a=release CLOSED TREE"
```

Comment:
This is a candidate to be defined at the task definition level.

Reply:
We may also want this configurable per-tree. Some times/trees we may want to block on closed tree; other times we want to `CLOSED TREE`.

Reply:
I'm leaning toward not letting the commit message be configurable, but letting CLOSED TREE/DONTBUILD be configurable.